plugins/entando-plugin-jacms/src/main/java/com/agiletec/plugins/jacms/apsadmin/resource/MultipleResourceAction.java

```diff
@@ -0,0 +1,340 @@
+            setFileDescriptions(fileDescr);
+            List<Category> resCategories = resource.getCategories();
+            for (int i = 0; i < resCategories.size(); i++) {
+                Category resCat = resCategories.get(i);
+                this.getCategoryCodes().add(resCat.getCode());
+            }
+            this.setMainGroup(resource.getMainGroup());
+            this.setStrutsAction(ApsAdminSystemConstants.EDIT);
+
+        } catch (Throwable t) {
+            logger.error("error in edit", t);
+            return FAILURE;
+        }
+        return SUCCESS;
+    }
+
+    protected void validateFileDescriptions() {
+        int fileUploadSize = 0;
+        if (null != this.getFileUpload()) {
+            fileUploadSize = this.getFileUpload().size();
+        }
+        for (int i = 0; i < fileUploadSize; i++) {
+            if (null != fileDescriptions) {
+                if (!(fileDescriptions.get(i).length() > 0)) {
+                    this.addFieldError(DESCR_FIELD + i, this.getText("error.resource.file.descrEmpty"));
+                }
+                if (fileDescriptions.get(i).length() > 250) {
+                    this.addFieldError(DESCR_FIELD + i, this.getText("error.resource.file.descrTooLong"));
+                }
+            } else {
+                this.addFieldError(DESCR_FIELD + i, this.getText("error.resource.file.descrEmpty"));
+            }
+        }
+    }
+
+    public String joinCategory() {
+        fetchFileDescriptions();
+        return super.joinCategory();
+    }
+
+    public String removeCategory() {
+        fetchFileDescriptions();
+        return super.removeCategory();
+    }
+
+    protected void checkRightFileType(ResourceInterface resourcePrototype, String fileName) {
+        boolean isRight = false;
+        if (fileName.length() > 0) {
+            String docType = fileName.substring(fileName.lastIndexOf('.') + 1).trim();
+            String[] types = resourcePrototype.getAllowedFileTypes();
+            isRight = isValidType(docType, types);
+        } else {
+            isRight = true;
+        }
+        if (!isRight) {
+            this.addFieldError("upload", this.getText("error.resource.file.wrongFormat"));
+        }
+    }
+
+    protected boolean isValidType(String docType, String[] rightTypes) {
+        boolean isValid = false;
+        if (rightTypes.length > 0) {
+            for (int i = 0; i < rightTypes.length; i++) {
+                if (docType.toLowerCase().equals(rightTypes[i])) {
+                    isValid = true;
+                    break;
+                }
+            }
+        } else {
+            isValid = true;
+        }
+        return isValid;
+    }
+
+    @Override
+    public String save() {
+        int index = 0;
+        savedId.clear();
+
+        boolean hasError = false;
+        try {
+
+            fetchFileDescriptions();
+            for (String fileDescription : getFileDescriptions()) {
```

Comment:
A lot going on here. Some debug level logging to help troubleshoot in the future would be helpful around the high level actions for each file.